src/Features/Core/Portable/EmbeddedLanguages/AbstractLanguageDetector.cs

```diff
@@ -93,34 +112,138 @@ public bool IsPossiblyPatternToken(SyntaxToken token, ISyntaxFacts syntaxFacts)
             return false;
         }
 
-        public bool IsEmbeddedLanguageString(SyntaxToken token, SemanticModel semanticModel, CancellationToken cancellationToken, out TOptions options)
+        public bool IsEmbeddedLanguageToken(SyntaxToken token, SemanticModel semanticModel, CancellationToken cancellationToken, out TOptions options)
         {
             options = default;
-            if (!IsPossiblyPatternToken(token, Info.SyntaxFacts))
-                return false;
 
             var syntaxFacts = Info.SyntaxFacts;
+            if (syntaxFacts.IsStringLiteral(token))
+                return IsEmbeddedLanguageStringLiteralToken(token, semanticModel, cancellationToken, out options);
+
+            if (token.RawKind == syntaxFacts.SyntaxKinds.InterpolatedStringTextToken)
+            {
+                options = default;
+                return IsEmbeddedLanguageInterpolatedStringTextToken(token, semanticModel, cancellationToken);
+            }
```

Comment:
added to support datetime lang support (which can appear in an interpolation format clause).  Not used by regex/json.